@here/harp-mapview/lib/copyrights/CopyrightProvider.ts

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { GeoBox } from "@here/harp-geoutils";
+import { getOptionValue } from "@here/harp-utils";
+import { CopyrightInfo } from "./CopyrightInfo";
+
+declare const require: any;
+
+// tslint:disable-next-line:no-var-requires
+const RBush = require("rbush");
+
+/**
+ * `CopyrightProvider` is an abstract class to retrieve copyrights information for geographic region
+ * specified by bounding box.
+ */
+export abstract class CopyrightProvider {
+    /** Predefined fixed HERE copyright info. */
+    static readonly HERE_COPYRIGHT_INFO: CopyrightInfo = {
```

Comment:
Since it was agreed to remove `DefaultCopyrightProvider` in lieu of default `copyrightInfo`, it will be useful to at least store this default HERE `CopyrightInfo` w/ current year somewhere for easier initialization. For instance, almost all examples have this default copyright. I suggest to keep it here as a static property of `CopyrightProvider`. If you have objections I will remove it from here.